src/workerd/api/sql.h

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include <workerd/jsg/jsg.h>
+#include <workerd/jsg/string.h>
+#include "basics.h"
+
+namespace workerd::api {
+
+typedef kj::OneOf<kj::Array<const byte>, kj::String, double> ValueBind;
+class SqlDatabase;
+
+class SqlResult final: public jsg::Object {
+public:
+  SqlResult(SqliteDatabase::Query& query);
+
+  kj::ArrayPtr<jsg::Dict<SqliteDatabase::Query::ValueOwned>> getResults(jsg::Lock& js);
+
+  JSG_RESOURCE_TYPE(SqlResult, CompatibilityFlags::Reader flags) {
+    JSG_READONLY_PROTOTYPE_PROPERTY(results, getResults);
+  }
+
+private:
+  kj::Vector<jsg::Dict<SqliteDatabase::Query::ValueOwned>> results;
+};
+
+class SqlPreparedStatement final: public jsg::Object {
+public:
+  SqlPreparedStatement(jsg::Ref<SqlDatabase>&& sqlDb, SqliteDatabase::Statement&& statement);
+
+  jsg::Ref<SqlResult> run(kj::Maybe<kj::Array<ValueBind>> bindValuesOptional);
+
+  JSG_RESOURCE_TYPE(SqlPreparedStatement, CompatibilityFlags::Reader flags) {
+    JSG_METHOD(run);
+  }
+
+  void visitForGc(jsg::GcVisitor& visitor) {
+    visitor.visit(sqlDatabase);
+  }
+
+private:
+  jsg::Ref<SqlDatabase> sqlDatabase;
+  SqliteDatabase::Statement statement;
+};
+
+class SqlDatabase final: public jsg::Object {
+public:
+  friend class SqlPreparedStatement;
+  SqlDatabase();
+
+  static jsg::Ref<SqlDatabase> constructor(jsg::Lock& js, CompatibilityFlags::Reader flags);
+  // Creates a new SqlDatabase.
+
+  jsg::Ref<SqlResult> exec(jsg::Lock& js, kj::String query, kj::Maybe<kj::Array<ValueBind>> bindValuesOptional);
+  jsg::Ref<SqlPreparedStatement> prepare(jsg::Lock& js, kj::String query);
+
+  JSG_RESOURCE_TYPE(SqlDatabase, CompatibilityFlags::Reader flags) {
+    JSG_METHOD(exec);
+    JSG_METHOD(prepare);
+  }
+
+private:
+  kj::Own<kj::Directory> dir;
```

Comment:
Is the intent for this to only work in workerd or also on the edge? If the latter, is using `kj::Directory` directly here the right thing?